### PR TITLE
tests: basic periph timer

### PR DESCRIPTION
**Pull request by smlng**
This PR is a refined and stripped down version of #30, i.e. adapting tests scripts and firmware to adhere with the structure and concept of existing tests.

### CODE REVIEW

**Comment by smlng** [comment]
@MichelRottleuthner as discussed f2f these are my refinements of your initial PR which you can use as starting point to work on more complex timer tests.

File: tests/periph_timer/tests/periph_timer.keywords.txt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    [Documentation]           Enable tracing for a gpio debug pin
+    API Call Should Succeed   PHiLIP.Write Reg  gpio[${pin}].mode.init  0
+    API Call Should Succeed   PHiLIP.Write Reg  gpio[${pin}].mode.io_type  3
+    API Call Should Succeed   PHiLIP.Execute Changes
+
+Timer Set To ${ticks} And Measure
+    [Documentation]           Set a timer and use PHiLIP to measure the delay
+    ...                       between calling set_timer and entering the timer
+    ...                       ISR callback by tracing a GPIO debug pin
+    Run Keywords              PHiLIP Reset
```
Comment (MrKevinWeiss) [inline tests/periph_timer/tests/periph_timer.keywords.txt]: Should the reset really be in the keyword?

File: tests/periph_timer/tests/periph_timer_if.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        cmds.append(self.timer_clear)
+        cmds.append(self.timer_read)
+        cmds.append(self.timer_start)
+        cmds.append(self.timer_stop)
+        cmds.append(self.timer_debug_pin)
+        cmds.append(self.timer_read_bench)
+        return cmds
+
+
+def main():
```
Comment (MrKevinWeiss) [inline tests/periph_timer/tests/periph_timer_if.py]: Should we keep doing this?

**Comment by smlng** [comment]
This one should be ready to go. Please note only the new boards support all features, i.e. the debug pins - so failing tests are expected. Alternative would be to remove the tests that require debug pins.

**Pull request merged**